src/GameSelector/index.tsx

```diff
@@ -0,0 +1,68 @@
+import { Button, Center, FormControl, Input, VStack } from "native-base";
+import React from "react";
+
+export interface GameSelectorFormData {
+  gameId: number | undefined;
+}
+
+const GameSelectorForm = () => {
+  const [formData, setData] = React.useState<GameSelectorFormData>({
+    gameId: undefined,
+  });
+  const [errors, setErrors] = React.useState({});
+
+  const validate = () => {
+    if (formData.gameId === undefined || formData.gameId === 0) {
+      setErrors({
+        ...errors,
+        gameId: "The game id cannot be empty",
+      });
+      return false;
+    }
+
+    setErrors({});
+    return true;
+  };
+
+  const onSubmit = () => {
+    validate() ? console.info("Submitted") : console.error("Validation Failed");
```

Comment:
Not a big fan ... As written, you don't necessarily expect that the `validate` method will change the state ... If you want  `console.log` for debug, put them in the `validate` method. Then you can write something simpler and more readable like `const onSubmit = () => validate`